client/shaders/nodes_shader/opengl_vertex.glsl

```diff
@@ -160,6 +163,9 @@ float disp_z;
 	normal = normalize(gl_NormalMatrix * gl_Normal);
 	tangent = normalize(gl_NormalMatrix * gl_MultiTexCoord1.xyz);
 	binormal = normalize(gl_NormalMatrix * gl_MultiTexCoord2.xyz);
+#ifdef SECONDSTAGE
+	normalPass = normalize((gl_Normal+1)/2);
```

Comment:
Am I right it is in the world space? Also why do you normalize the value you scale?

Reply:
It is in the world space. The normalization is probably not needed.

Reply:
Just realized the normalization is harmful here. E.g. if `gl_Normal` is either (−1/√3, −1/√3, −1/√3) or (1/√3, 1/√3, 1/√3) `normalPass` will be (1/√3, 1/√3, 1/√3) in both cases. So you may normalize *before* adding one, but never after.